xwiki-platform-core/xwiki-platform-rest/xwiki-platform-rest-server/src/main/java/org/xwiki/rest/internal/ModelFactory.java

```diff
@@ -977,6 +979,33 @@ private static String serializePropertyValue(PropertyInterface property)
         }
     }
 
+    /**
+     * Serializes the value of the given XObject property. In case the property is an instance of
+     * {@link ComputedFieldClass}, the serialized value is the computed property value.
+     * @param property an XObject property
+     * @param propertyClass the PropertyClass of that XObject proprety
+     * @param context the XWikiContext
+     * @return the String representation of the property value
+     */
+    private static String serializePropertyValue(PropertyInterface property,
+        com.xpn.xwiki.objects.classes.PropertyClass propertyClass, XWikiContext context)
+    {
+        if (propertyClass instanceof ComputedFieldClass) {
+            // TODO: the XWikiDocument needs to be explicitely set in the context, otherwise method
+            // PropertyClass.renderInContext fires a null pointer exception: bug?
+            XWikiDocument document = context.getDoc();
+            context.setDoc(property.getObject().getOwnerDocument());
+            ComputedFieldClass computedFieldClass = (ComputedFieldClass) propertyClass;
+            String value = computedFieldClass.displayView(propertyClass.getName(), property.getObject(), context);
```

Comment:
A general issue is that displayView is designed to return HTML so this will work only for very simple String values. Same issue than when you use it in a script.

Reply:
Indeed... What about introducing a method ComputedFieldClass:getValue (or any similar name) which would execute the current code in ComputedFieldClass:displayView, and make sure that the ComputedFieldClass:displayView actually uses any possible custom displayer when there is one (it seems it's not the case at the moment), so that: "getValue" returns a raw computed value, and displayView returns the view to be displayed in HTML?

Reply:
> What about introducing a method ComputedFieldClass:getValue

+1 for some ComputedFieldClass:getComputedValue, it's a good idea since we know we are manipulating a ComputedFieldClass

Reply:
I plan to work on that, then. Would it make sense to you also to possibly add the "display" property to the REST output (it could be optional)? In some cases indeed, we want to display the HTML view of a property value in a different context than XWiki, or even within XWiki but loaded via an ajax call. What do you think? (that would be in a dedicated separate issue in any case)

Reply:
Not sure this is the right place for this need. I feel REST should be about data as much as possible. But we could think about some display resource handler.